corehq/ex-submodules/pillowtop/pillow/interface.py

```diff
@@ -161,21 +161,28 @@ def process_changes(self, since):
         # Not sure why this is need, but I'm preserving the behavior
         since = since or None
 
-        def process_offset_chunk(chunk, context):
-            if not chunk:
-                return
-            self._batch_process_with_error_handling(chunk)
-            # update checkpoint for just the latest change
-            self._update_checkpoint(chunk[-1], context)
+        def process_offset_chunk(chunk, context, last_change):
+            if chunk:
+                self._batch_process_with_error_handling(chunk)
+            # only attempt to update checkpoint if there is a latest change
+            if last_change:
+                self._update_checkpoint(last_change, context)
 
         # keep track of chunk for batch processors
         changes_chunk = []
         last_process_time = datetime.utcnow()
 
+        change = None  # meaning no changes seen yet
```

Comment:
Calling this something like `last_truthy_change` would make the code read a bit more clearly and let you reduce the comments.

Reply:
Yeah nice point. I'll try and clean this up in a followup